### ISSUE TITLE
how to solve large amount of query result ?

### ISSUE THREAD

**Issue by Tasselmi** [blynkkk/clickhouse4j#66]
how to solve large amount of query result ?
my query result is huge, and get a `java.lang.OutOfMemoryError
`, how can I solve it?

```
java.lang.OutOfMemoryError
  at java.io.ByteArrayOutputStream.hugeCapacity(ByteArrayOutputStream.java:123)
  at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:117)
  at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
  at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
  at cc.blynk.clickhouse.util.guava.StreamUtils.copy(StreamUtils.java:70)
  at cc.blynk.clickhouse.http.DefaultHttpConnector.sendPostRequest(DefaultHttpConnector.java:182)
  at cc.blynk.clickhouse.http.DefaultHttpConnector.post(DefaultHttpConnector.java:62)
  at cc.blynk.clickhouse.http.DefaultHttpConnector.post(DefaultHttpConnector.java:74)
  at cc.blynk.clickhouse.ClickHouseStatementImpl.sendRequest(ClickHouseStatementImpl.java:674)
  at cc.blynk.clickhouse.ClickHouseStatementImpl.executeQuery(ClickHouseStatementImpl.java:168)
  at cc.blynk.clickhouse.ClickHouseStatementImpl.executeQuery(ClickHouseStatementImpl.java:150)
  at cc.blynk.clickhouse.ClickHouseStatementImpl.executeQuery(ClickHouseStatementImpl.java:137)
  at cc.blynk.clickhouse.ClickHouseStatementImpl.executeQuery(ClickHouseStatementImpl.java:88)
  ... 59 elided
```

Thanks very much.

**Comment by doom369** [comment]
@Tasselmi hello, look like a bug to me. We will try to fix.

For now you can do the next:

1. Increase RAM;
2. Split the the input on batches by `limit` / `offset` parameters;
3. You can try CopyManager if that fits your needs;
4.Try to fallback to the original driver and check if it works with it - https://github.com/ClickHouse/clickhouse-jdbc

**Comment by doom369** [comment]
@Tasselmi I did a quick fix - https://github.com/blynkkk/clickhouse4j/releases/download/clickhouse4j-1.3.0/clickhouse4j-1.3.1-SNAPSHOT.jar

Could you please check with it?

**Comment by Tasselmi** [comment]
> @Tasselmi I did a quick fix - https://github.com/blynkkk/clickhouse4j/releases/download/clickhouse4j-1.3.0/clickhouse4j-1.3.1-SNAPSHOT.jar
> 
> Could you please check with it?

ok, let me have a try.

**Comment by Tasselmi** [comment]
> @Tasselmi I did a quick fix - https://github.com/blynkkk/clickhouse4j/releases/download/clickhouse4j-1.3.0/clickhouse4j-1.3.1-SNAPSHOT.jar
> 
> Could you please check with it?

It works fine now.

**Comment by doom369** [comment]
@Tasselmi cool. We just released new version 1.4.0 version. You can already use it within your dependency.

**Comment by Tasselmi** [comment]
> @Tasselmi cool. We just released new 1.4.0 version. You can already use it within your dependency.

thanks a lot.
